app/Platform/Services/UserAgentService.php

```diff
@@ -15,6 +15,13 @@ public function decode(string $userAgent): array
 
     private function parseUserAgent(string $userAgent): array
     {
+        if (empty($userAgent) || $userAgent === '[not provided]') {
```

Comment:
Special handling for [when a heartbeat call is made without a User Agent](https://github.com/RetroAchievements/RAWeb/blob/master/app/Platform/Events/PlayerSessionHeartbeat.php#L35). Also, short-circuit for null/empty entries that were previously being handled indirectly later.